grid_problem.py

```diff
@@ -2,19 +2,28 @@
 import math
 import random
 
-W = 10
-H = 12
-NUM_DISTRICTS = 7
+W = 20
```

Comment:
We should keep the committed versions of these smaller, right now hill climbing takes 12s with these numbers for me.